assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/dinov2/dinov2_embeddings_mlflow_wrapper.py

```diff
@@ -0,0 +1,93 @@
+    ) -> None:
+        """Initialize wrapper class.
+
+        :param task_type: Task type used for inference.
+        :type task_type: str
+        """
+        super().__init__()
+
+        self._task_type = task_type
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """
+        Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        from vision_utils import get_current_device
+
+        if self._task_type == Tasks.EMBEDDINGS.value:
+            try:
+                model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+                self._processor = AutoImageProcessor.from_pretrained(model_dir)
+                self._model = AutoModel.from_pretrained(model_dir)
+                self._device = get_current_device()
+                self._model.to(self._device)
+
+                print("Model loaded successfully")
+
+            except Exception as e:
+                print("Failed to load the the model.")
+                print(e)
+                raise
+
+        else:
+            raise ValueError(f"invalid task type {self._task_type}")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data: input images, either as publicly accessible urls or base64 strings
+        :type input_data: Pandas DataFrame with column "image"
+        :return: image embeddings
+        :rtype: Pandas DataFrame with column "image_features"
+        """
+        from vision_utils import process_image
+
+        # Initialize the result embedding list to empty.
+        embeddings = []
+
+        # Go through the input images.
+        for image in input_data[MLflowSchemaLiterals.INPUT_COLUMN_IMAGE]:
+            # Decode the image and make a PIL Image object.
+            pil_image = Image.open(io.BytesIO(process_image(image)))
+
+            # Do inference and extract the embeddings out of the output structure.
+            inputs = self._processor(images=pil_image, return_tensors="pt")
+            outputs = self._model(**{
+                key: inputs[key].to(self._device) for key in inputs
+            })
+            embedding = outputs.last_hidden_state[0, 0, :]
+
+            # Offload to CPU, convert to Python list and accumulate into result embedding list.
+            embedding = embedding.detach().cpu().numpy().tolist()
+            embeddings.append(embedding)
+
+        # Convert embeddings to Pandas dataframe and return.
+        df_responses = pd.DataFrame({MLflowSchemaLiterals.OUTPUT_COLUMN_IMAGE_FEATURES: embeddings})
+        return df_responses
```

Comment:
Just clarification, are we not going to support batch deployment for this?

Reply:
Guessing you are referring to the absence of `run_inference_batch()`? AFAIU from Phani, this is just a utility function that is not called elsewhere, so batch deployment should be independent of this. If you know more about the code, please comment.